docs/components/Combobox/Combobox.stories.mdx

```diff
@@ -47,6 +47,16 @@ handling user interactions, and managing search functionality.
 Actions are an optional way to allow users to perform custom tasks such as
 adding a new item to the list of options.
 
+#### Callback
+
+`onSelect` is the recommended callback handler that should be used in most cases. This will immediately inform
+the consumer of changes to selections as they happen.
+
+If however immediate changes are not desired, and instead it is preferable to wait until all selections or
+modifications to selections have been made before informing the consumer, `onClose` may be used. The arguments provided to this callback are
+Identical to those of `onSelect`, only the timing differs. Please note this will only
+have a noticeable effect with `multiSelect = true` since the Combobox closes immediately upon selection with single-select.
```

Comment:
I feel conflicted about this

I never like it when two props are incompatible, or one does nothing when the other is present. if they are incompatible I prefer to do something like XOR where you simply can't combine them if we know they aren't valid

in this case though, technically speaking `onClose` isn't incompatible, nor does not do anything. in fact it does the exact same thing as `onSelect` since for single select it closes when you select. so my question is, is this fine? do we need to do anything different? it's just odd that you have two different callbacks that do the same thing for single select

Reply:
interesting idea. the thing that wouldn't offer is the flexibility to do something on select, and then also do something on close. it would be a one or the other with the boolean. which is admittedly the boat we're in right now, but I am reconsidering!

looking at some other implementations, they also have an `onChange` and an `onClose`

![image](https://github.com/GetJobber/atlantis/assets/11843215/c86bca9f-e278-43e9-82b5-d3068544f9ec)

the difference being that their `onClose` doesn't have the selections as an argument, but you're always free to ignore those.